lib/plugins/aws/deploy/index.js

```diff
@@ -126,6 +124,10 @@ class AwsDeploy {
         mainProgress.notice('Retrieving CloudFormation stack', { isMainEvent: true }),
       'aws:deploy:deploy:createStack': async () => this.createStack(),
 
+      'before:aws:deploy:deploy:ensureValidBucketExists': () =>
+        mainProgress.notice('Ensuring that deployment bucket exists', { isMainEvent: true }),
```

Comment:
I'm not sure we need a new progress event here. 

It feels to me as part of _Retrieving CloudFormation stack_ part, and it's pretty fast, so unlikely to be noticeable

Reply:
I wasn't sure how to capture that properly, but I like your suggestions around `progress` 👍